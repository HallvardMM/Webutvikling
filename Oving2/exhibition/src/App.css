.App {
  text-align: center;
  display: grid;
  grid-template-columns: [firstColumn] 10% [secondColumn] 80% [thirdColumn] 10%;
}

.mainTitle {
  font-size: 1.75rem;
  margin-bottom: 0%;
}

.about {
  font-size: 1.5rem;
}

.text {
  font-size: 1rem;
}

.DropDown {
  grid-column: firstColumn;
}

.Main {
  grid-column: secondColumn;
}

ul {
  list-style-type: none;
}

*:focus {
  outline: 0 !important;
}


@media(max-width: 550px){
  .mainTitle{
    font-size: 1rem;
  }
  .about {
    font-size: 1rem;
  }
}