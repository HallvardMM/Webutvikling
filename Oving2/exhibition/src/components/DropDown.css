.DropDown {
  display: flex;
  flex-direction: column;
  margin-left: 1vh;
  margin-top: 2vh;
  align-items: flex-start;
}

.settingsButton {
  border-radius: 4px;
  padding: 5px 5px 0px 5px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin: 4px 20px;
  transition-duration: 0.4s;
  cursor: pointer;
  border-width: 2px;
  border-style: solid;
}

.settingsButton:hover {
  background-color: whitesmoke !important;
}

@media (max-width: 1024px) {
  .settingsButton {
    padding: 4.5px 4.5px 0px 4.5px;
  }
}

@media (max-width: 768px) {
  .settingsButton {
    padding: 3.5px 3.5px 0px 3.5px;
  }
}

@media (max-width: 500px) {
  .settingsButton {
    padding: 2px 2px 0px 2px;
    margin: 0;
  }
  .DropDown {
    margin-left: 0;
    margin-top: 0;
  }
}
