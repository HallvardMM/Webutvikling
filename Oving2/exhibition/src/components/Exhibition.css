.Exhibition {
  height: 100%;
  display: flex;
    flex-direction: column;
    align-items: center;
}

.row {
  display: flex;
  justify-content: center;
}

.columnRight {
  display: flex;
  flex-direction: column;

  width: 10%;
}

.ExhibitionbuttonLeft,
.ExhibitionbuttonRight {
  margin-top: auto;
  margin-bottom: auto;
  border-radius: 4px;
  padding: 2px 4px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  transition-duration: 0.4s;
  cursor: pointer;
  border-width: 2px;
  border-style: solid;
}

.ExhibitionbuttonLeft:hover,
.ExhibitionbuttonRight:hover,
.ExtraButton:hover {
  background-color: whitesmoke !important;
}

.ExhibitionbuttonLeft {
  align-self: flex-end;
}

.ExtraButton {
  border-radius: 4px;
  padding: 5px 5px 0px 5px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin: 4px 20px;
  transition-duration: 0.4s;
  cursor: pointer;
  border-width: 2px;
  border-style: solid;
}
.ExtraButtons {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.ExtraButtons div{
  display: flex;
    flex-direction: column;
}

@media (max-width: 1024px) {
  .ExhibitionbuttonLeft,
  .ExhibitionbuttonRight {
    font-size: 0.75rem;
  }
  .ExtraButton {
    padding: 4.5px 4.5px 0px 4.5px;
  }
}

@media (max-width: 768px) {
  .ExhibitionbuttonLeft, .ExhibitionbuttonRight {
    font-size: 0.7rem;
  }
  .ExtraButton {
    padding: 3.5px 3.5px 0px 3.5px;
  }
}

@media (max-width: 550px) {
  .ExhibitionbuttonLeft{
    margin-right: 5px;
  }
  .ExhibitionbuttonRight{
    margin-left: 5px;
  }

  .ExtraButton {
    padding: 2px 2px 0px 2px;
  }
}
