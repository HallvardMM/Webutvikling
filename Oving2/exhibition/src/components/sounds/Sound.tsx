import React from 'react'
const Abstract = require("./Abstract.mp3");
const Car = require("./Car.mp3");
const Desert = require("./Desert.mp3");
const Duff = require("./Duff.mp3");
const Head = require("./Head.mp3");
const House = require("./House.mp3");
const ProgPils = require("./ProgPils.mp3");
const TwoHouses = require("./TwoHouses.mp3");

interface Musicoptions {
    Option: typeof Abstract,
    Mute: boolean
}

export const songs = {
    Abstract: Abstract,
    Car: Car,
    Desert: Desert,
    Duff: Duff,
    Head: Head,
    House: House,
    ProgPils: ProgPils,
    TwoHouse: TwoHouses
}

export default function Sound(props: Musicoptions) {
    return (
        <div>
            <audio src={props.Option} autoPlay loop muted={props.Mute} />
        </div>
    );
}
