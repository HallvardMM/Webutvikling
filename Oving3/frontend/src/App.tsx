import React from 'react';
import { Route, Switch } from 'react-router-dom'
import Main from './maincomponents/Main';
import Registration from './maincomponents/Registration';
import { makeStyles, createStyles } from '@material-ui/core/styles';
import {useSelector} from 'react-redux';
import { selectTable } from './reducers/themeReducer';


//App is a SPA with support for routing between registration page and main

function App() {
  const tableBackground = useSelector(selectTable);
  const useStyles = makeStyles(() =>
      createStyles({
          paper: {
              background: tableBackground,
              height: '100vh',
              width: 'auto',
          }
      }));
  const classes = useStyles();
  return (
    <div className={classes.paper}>
      <Switch>
        <Route exact path="/" component={Main} />
        <Route path="/registration" component={Registration} />
      </Switch>
    </div>
  );
}

export default App;
