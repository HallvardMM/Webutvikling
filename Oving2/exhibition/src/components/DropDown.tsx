import React, { useState, useContext, CSSProperties } from 'react';
import DropDownMenu from './Settings';
import { ReactComponent as Settings } from './icons/settings.svg';
import { ThemeContext } from '../context/ThemeContext';
import './DropDown.css';

export default function Dropdown() {
  const [dropDownOpen, setDropDownOpen] = useState(false);
  const context = useContext(ThemeContext);

  const settingsButtonColor: CSSProperties = {
    backgroundColor: context.Theme.BoxColor,
    color: context.Theme.FrameColor,
    borderColor: context.Theme.FrameColor,

  }

  return (
    <div className='DropDown'>
      <div>
        <button className="settingsButton" style={settingsButtonColor} onClick={() => setDropDownOpen(!dropDownOpen)}><Settings></Settings></button>
      </div>
      <div>{dropDownOpen ? <DropDownMenu /> : <div />}</div>
    </div>
  );
}
