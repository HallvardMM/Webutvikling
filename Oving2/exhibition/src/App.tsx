import React, { useState } from 'react';
import './App.css';
import DropDown from './components/DropDown';
import Exhibition from './components/Exhibition';
import { ThemeContext, Theme } from './context/ThemeContext';
import StorageKeysContext from './context/StorageContext';

export default function App() {
  //Get theme and set theme state used to set state in context
  const [theme, setTheme] = useState(Theme.Blue);

  return (
    <ThemeContext.Provider value={{ Theme: theme, update: setTheme }}>
      <ThemeContext.Consumer>
        {(value) => (
          <div
            className='App'
            id='App'
            style={{ backgroundColor: value.Theme.BackgroundColor }}
          >
            <StorageKeysContext.Provider
              value={['favoritArt', 'speedOfRotation']}
            >
              <div className='DropDown'>
                <DropDown></DropDown>
              </div>
              <div className='Main'>
                <h1 className='mainTitle'>Worlds Greatest Art Exhibition!</h1>
                <Exhibition></Exhibition>
                <h2 className='about'>About the exhibition</h2>

                <h4 className='text'>Info</h4>

                <p className='text'>
                  This exhibition is made to combine different artistic mediums
                  to a large art piece. The exhibition contains eight art
                  pieces with a picture made in SVG, a title, sound and the
                  first line from a poem randomly collected from{' '}
                  <a
                    href='https://poetrydb.org/index.html'
                    rel='noopener noreferrer'
                    target='_blank'
                  >
                    poetryDB.
                  </a>
                </p>

                <h4 className='text'>Additions</h4>

                <p className='text'>
                  By user request we added the possibility to save a art piece.
                  You can do so by favoriting an image by clicking the star
                  button. Next time you look at the picture the star button is
                  marked and the poem is the same as the saved one. If you unfavorit a new poem wil be added next time.
                </p>
                <p className='text'>
                  Rotation is default 10s and can be changed in settings. The change will be saved until you close your browser.
                  Apply be pressing enter after writing in a number from 1s to how high you want in seconds.
                  You can pause with pause-button and it will automatically pause if you change picture manually.
                </p>
                <p className='text'>
                  In settings you can change color theme and it will stay the same until you refresh the page.
                </p>
                <p className='text'>
                  It is added fitting music to the art. If you want to enjoy the art in silence you can press the sound-button and mute.
                </p>

                <ul>
                  <b className='text'>Credits for the music and sound</b>
                  <li className='text'>
                    Abstract: Piano Horror - Francisco Alvear{' '}
                    <a
                      href='https://mixkit.co/free-stock-music/'
                      rel='noopener noreferrer'
                      target='_blank'
                    >
                      Link to song
                    </a>
                  </li>
                  <li className='text'>
                    Car: Car sounds from 1:00 - 1:08{' '}
                    <a
                      href='https://www.youtube.com/watch?v=hWrwloX6H9U&ab_channel=19Bozzy92'
                      rel='noopener noreferrer'
                      target='_blank'
                    >
                      Link to sounds
                    </a>
                  </li>
                  <li className='text'>
                    Desert: Sun and His Daughter - Eugenio Mininni{' '}
                    <a
                      href='https://mixkit.co/free-stock-music/'
                      rel='noopener noreferrer'
                      target='_blank'
                    >
                      Link to song
                    </a>
                  </li>
                  <li className='text'>
                    Duff: Comical - Ahjay Stelino{' '}
                    <a
                      href='https://mixkit.co/free-stock-music/'
                      rel='noopener noreferrer'
                      target='_blank'
                    >
                      Link to song
                    </a>
                  </li>
                  <li className='text'>
                    Head: Hip Hop 02 - Lily J{' '}
                    <a
                      href='https://mixkit.co/free-stock-music/'
                      rel='noopener noreferrer'
                      target='_blank'
                    >
                      Link to song
                    </a>
                  </li>
                  <li className='text'>
                    House: Spirit in the Woods - Alejandro Maga√±a (A. M.){' '}
                    <a
                      href='https://mixkit.co/free-stock-music/'
                      rel='noopener noreferrer'
                      target='_blank'
                    >
                      Link to song
                    </a>
                  </li>
                  <li className='text'>
                    ProgPils: Games Worldbeat - Bernardo R.{' '}
                    <a
                      href='https://mixkit.co/free-stock-music/'
                      rel='noopener noreferrer'
                      target='_blank'
                    >
                      Link to song
                    </a>
                  </li>
                  <li className='text'>
                    TwoHouse: Sugar Baby - Dock Boggs{' '}
                    <a
                      href='http://www.openmusicarchive.org/search.php'
                      rel='noopener noreferrer'
                      target='_blank'
                    >
                      Link to song
                    </a>
                  </li>
                </ul>
              </div>
            </StorageKeysContext.Provider>
          </div>
        )}
      </ThemeContext.Consumer>
    </ThemeContext.Provider>
  );
}
