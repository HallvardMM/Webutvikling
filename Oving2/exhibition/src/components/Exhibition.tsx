import React, { useState, useEffect, useContext, CSSProperties } from 'react';
import './Exhibition.css';

import Art from './Art';
import Sound, { songs } from './sounds/Sound';
import storageKeysContext from '../context/StorageContext';
import { ThemeContext } from '../context/ThemeContext';

import { ReactComponent as TwoHouses } from '../Pictures/twoHouses.svg';
import { ReactComponent as Desert } from '../Pictures/desert.svg';
import { ReactComponent as House } from '../Pictures/house.svg';
import { ReactComponent as Head } from '../Pictures/head.svg';
import { ReactComponent as Racecar } from '../Pictures/racecar.svg';
import { ReactComponent as Abstract } from '../Pictures/abstract.svg';
import { ReactComponent as ProgPils } from '../Pictures/progPils.svg';
import { ReactComponent as Duff } from '../Pictures/duff.svg';
import { ReactComponent as Pause } from './icons/pause.svg';
import { ReactComponent as Play } from './icons/play.svg';
import { ReactComponent as Star } from './icons/star.svg';
import { ReactComponent as UnStar } from './icons/unstar.svg';
import { ReactComponent as Audio } from './icons/audio.svg';
import { ReactComponent as Mute } from './icons/mute.svg';

import '../Pictures/twoHouses.css';
import '../Pictures/desert.css';
import '../Pictures/house.css';
import '../Pictures/head.css';
import '../Pictures/racecar.css';
import '../Pictures/abstract.css';
import '../Pictures/progPils.css';
import '../Pictures/duff.css';

interface ArtPiece {
    Title: string;
    Picture: React.FunctionComponent<React.SVGProps<SVGSVGElement>>;
    Music: typeof songs.Abstract;
}

const TheArt: Record<number, ArtPiece> = {
    /*Record with the art installations */
    0: { Title: 'Abstract', Picture: Abstract, Music: songs.Abstract },
    1: { Title: 'Car', Picture: Racecar, Music: songs.Car },
    2: { Title: 'Desert', Picture: Desert, Music: songs.Desert },
    3: { Title: 'Duff', Picture: Duff, Music: songs.Duff },
    4: { Title: 'Head', Picture: Head, Music: songs.Head },
    5: { Title: 'House', Picture: House, Music: songs.House },
    6: { Title: 'ProgPils', Picture: ProgPils, Music: songs.ProgPils },
    7: { Title: 'TwoHouses', Picture: TwoHouses, Music: songs.TwoHouse },
};

const noOutOfBounds = (number: number) => {
    /* Checks if picture is outside of range and creates a sirkling effect */
    switch (number) {
        case -1:
            return 7;
        case 8:
            return 0;
    }
    return number;
};

const useStateWithLocalStorage = (localStorageKey: string) => {
    /*Function for the logic of saving favorites as array as string to localStorage */
    const [value, setValue] = useState(
        JSON.parse(localStorage.getItem(localStorageKey) || '[]')
    );
    useEffect(() => {
        const oldData = localStorage.getItem(localStorageKey);
        if (oldData) {
            const jsonOldData = JSON.parse(oldData);
            if (!Array.isArray(value[0])) {
                /*Don't add old data from LocalStorage */
                localStorage.setItem(
                    localStorageKey,
                    JSON.stringify([...jsonOldData, value])
                );
            }
        } else {
            localStorage.setItem(localStorageKey, JSON.stringify([]));
        }
    }, [value]);

    return [value, setValue];
};

const removeFromLocalStorage = (localStorageKey: string, remove: string) => {
    /*Function for the logic of removing favorites as array as string to localStorage */
    const data = JSON.parse(localStorage.getItem(localStorageKey) || '[]');
    for (let i in data) {
        if (data[i][0] === remove) {
            data.splice(i, 1);
            localStorage.setItem(localStorageKey, JSON.stringify(data));
        }
    }
};

const isFavorited = (localStorageKey: string, artNumber: number) => {
    /*Function to check if picture is in localStorage as a favorit */
    const data = JSON.parse(localStorage.getItem(localStorageKey) || '[]');
    for (let i in data) {
        if (data[i][0] === artNumber.toString()) {
            return true;
        }
    }
    return false;
};

const getPoetry = (localStorageKey: string, artNumber: number) => {
    /*Function to fetch saved poetry form localstorage */
    const data = JSON.parse(localStorage.getItem(localStorageKey) || '[]');
    for (let i in data) {
        if (data[i][0] === artNumber.toString()) {
            return data[i];
        }
    }
    return false;
};

const speedSessionStorage = (sessionStorageKey: string) => {
    /*Fetching rotation speed from sessionStorage and returning number of seconds
      if user tries to input less then 1 sec return 10 sec */
    const speedString = sessionStorage.getItem(sessionStorageKey);
    if (speedString) {
        if (Number(speedString) >= 1) {
            return Number(speedString);
        } else {
            return 10;
        }
    } else {
        return 10;
    }
};

export default function Exhibition() {
    const StorageKeys = useContext(
        storageKeysContext
    ); /*[Local,Session] StorageKey from context api */
    const [playing, setPlaying] = useState(
        true
    ); /*state for pause and play of auto scrolling of pictures*/
    const [mute, setMute] = useState(false); /*state for muting music*/
    const [art] = useState(
        TheArt
    ); /*The art saved as a state TODO: add logic for changing to favorits*/
    const [artNumber, setArtNumber] = useState(
        0
    ); /*Which art piece is showing numbers are from "TheArt" record*/
    const [star, setStar] = useState(
        isFavorited(StorageKeys[0], artNumber)
    ); /*State to save if the art is favorited */
    const [error, setError] = useState(
        false
    ); /*Error state if loading for Poetry failes */
    const [isLoaded, setIsLoaded] = useState(
        false
    ); /* Don't show component before poetry is loaded*/
    const [poetry, setPoetry] = useState(
        ''
    ); /*Sets the poetry fetched from PoetryDB*/
    const [poetryTitle, setPoetryTitle] = useState(
        ''
    ); /*Sets the title fetched from PoetryDB*/
    const [poetryAuthor, setPoetryAuthor] = useState(
        ''
    ); /*Sets the author fetched from PoetryDB*/
    const [favorit, setFavorit] = useStateWithLocalStorage(
        StorageKeys[0]
    ); /*Sets the favorited poems*/
    const [TimeoutID, setTimeoutID] = useState(
        -1
    ); /*Time for next rotation of art -1 dummy value */
    const context = useContext(ThemeContext); /* ColorTheme set and get */

    const ButtonColor: CSSProperties = {
        backgroundColor: context.Theme.BoxColor,
        borderColor: context.Theme.FrameColor,
        color: context.Theme.FrameColor,
    };

    const FrameColor: CSSProperties = {
        borderColor: context.Theme.FrameColor,
    };

    useEffect(() => {
        /*Each time the artpiece changes fetch poetry */
        const savedPoetry = getPoetry(StorageKeys[0], artNumber);
        if (!savedPoetry) {
            /*Poetry not saved in LocalStorage fetch random poetry */
            fetch('https://poetrydb.org/random')
                .then((res) => res.json())
                .then(
                    (result) => {
                        if (Array.isArray(result)) {
                            //PoetryDB added breaking changes to return of random New return array
                            setIsLoaded(true);
                            setPoetry(result[0].lines[0]);
                            setPoetryTitle(result[0].title);
                            setPoetryAuthor(result[0].author);
                        } else {
                            //PoetryDB added breaking changes to return of random old return object
                            setIsLoaded(true);
                            setPoetry(result.lines[0]);
                            setPoetryTitle(result.title);
                            setPoetryAuthor(result.author);
                        }
                    },
                    (error) => {
                        setIsLoaded(true);
                        setError(true);
                        console.error(error);
                    }
                );
        } else {
            /*Poetry saved in LocalStorage fetch this poetry*/
            setPoetry(savedPoetry[3]);
            setPoetryAuthor(savedPoetry[2]);
            setPoetryTitle(savedPoetry[1]);
            setIsLoaded(true);
        }
    }, [artNumber]);

    useEffect(() => {
        /* Checks if picture is in favorited in localstorage and setStar*/
        setStar(isFavorited(StorageKeys[0], artNumber));
    }, [artNumber]);

    useEffect(() => {
        /* For keeping favorit updated when favoriting a picture */
        setFavorit(favorit);
    }, [star]);

    function changeArt(number: number) {
        /* Used when changing picture.*/
        setArtNumber(noOutOfBounds(number));
        setIsLoaded(false);
    }

    function changeArtManual(number: number) {
        /* When clicking next manually stop rotation*/
        changeArt(number);
        setPlaying(false);
    }

    const addFavorit = (isFavorit: Boolean, artNr: number) => {
        /* Add or remove picture from favorits*/
        if (isFavorit) {
            setFavorit([artNumber.toString(), poetryTitle, poetryAuthor, poetry]);
        } else {
            removeFromLocalStorage(StorageKeys[0], artNr.toString());
        }
        setStar(!star);
    };

    useEffect(() => {
        /*Triggers setTimeout when clicking on the Autoplay-button*/
        if (playing) {
            setTimeoutID(
                window.setTimeout(
                    () => changeArt(artNumber + 1),
                    speedSessionStorage(StorageKeys[1]) * 1000
                )
            );
        } else {
            window.clearTimeout(TimeoutID);
        }
    }, [playing, artNumber]);

    return (
        <div className='Exhibition'>
            <div className='row'>
                <button
                    className='ExhibitionbuttonLeft'
                    style={ButtonColor}
                    onClick={() => changeArtManual(artNumber - 1)}
                >
                    &lt; Prev
          </button>
                <Art
                    error={error}
                    isLoading={!isLoaded}
                    Title={art[artNumber].Title}
                    Poetry={poetry}
                    PoetryAuthor={poetryAuthor}
                    PoetryTitle={poetryTitle}
                    Picture={art[artNumber].Picture}
                    FrameColor={FrameColor}
                ></Art>
                <button
                    className='ExhibitionbuttonRight'
                    style={ButtonColor}
                    onClick={() => changeArtManual(artNumber + 1)}
                >
                    Next &gt;
          </button>
            </div>
            <div className='ExtraButtons'>
                <div>
                    <button
                        className='ExtraButton'
                        style={ButtonColor}
                        onClick={() => setPlaying(!playing)}
                    >
                        {' '}
                        {playing ? <Pause></Pause> : <Play></Play>}{' '}
                    </button>
                    Rotation
                </div>
                <div>
                    <button
                        className='ExtraButton'
                        style={ButtonColor}
                        onClick={() => addFavorit(!star, artNumber)}
                    >
                        {star ? <Star></Star> : <UnStar></UnStar>}{' '}
                    </button>
                    Favorite
                </div>
                <div>
                    <button
                        className='ExtraButton'
                        style={ButtonColor}
                        onClick={() => setMute(!mute)}
                    >
                        {mute ? <Mute></Mute> : <Audio></Audio>}{' '}
                    </button>
                    Sound
                </div>
            </div>
            <Sound Option={art[artNumber].Music} Mute={mute}></Sound>
        </div>
    );
}
