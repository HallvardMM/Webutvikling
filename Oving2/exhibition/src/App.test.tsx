import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';
import DropDownMenu from './components/Settings';
import Exhibition from './components/Exhibition';
import Sound from './components/sounds/Sound';
import DropDown from './components/DropDown';

test('renders a snapshot', () => {
  //check if the app runs
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly dropdownmenu', () => {
  //check if dropdownmenu exists
  const dropdownmenu = renderer.create(<DropDownMenu />).toJSON();
  expect(dropdownmenu).toBeTruthy;
});

test('render correctly exhibition', () => {
  //check if exhibition exists
  const exhibition = renderer.create(<Exhibition />).toJSON();
  expect(exhibition).toBeTruthy;
});

test('render correctly dropdown', () => {
  //check if dropdown exists
  const dropdown = renderer.create(<DropDown />).toJSON();
  expect(dropdown).toBeTruthy;
});

test('render correctly sound', () => {
  //check if sound renders
  const Abstract = require('./components/sounds//Abstract.mp3');
  const sound = renderer
    .create(<Sound Option={Abstract} Mute={false} />)
    .toJSON();
  expect(sound).toBeTruthy;
});
