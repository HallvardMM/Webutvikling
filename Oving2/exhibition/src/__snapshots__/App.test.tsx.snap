// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="App"
  id="App"
  style={
    Object {
      "backgroundColor": "cornflowerblue",
    }
  }
>
  <div
    className="DropDown"
  >
    <div
      className="DropDown"
    >
      <div>
        <button
          className="settingsButton"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "lightskyblue",
              "borderColor": "mediumblue",
              "color": "mediumblue",
            }
          }
        >
          <svg>
            settings.svg
          </svg>
        </button>
      </div>
      <div>
        <div />
      </div>
    </div>
  </div>
  <div
    className="Main"
  >
    <h1
      className="mainTitle"
    >
      Worlds Greatest Art Exhibition!
    </h1>
    <div
      className="Exhibition"
    >
      <div
        className="row"
      >
        <button
          className="ExhibitionbuttonLeft"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "lightskyblue",
              "borderColor": "mediumblue",
              "color": "mediumblue",
            }
          }
        >
          &lt; Prev
        </button>
        <div
          className="Art"
        >
          <h2
            className="ArtTitle"
          >
            Title: 
            Abstract
          </h2>
          <div
            className="poetryDiv"
          >
            <p>
              Poetry is loading
            </p>
          </div>
          <div
            className="ArtFrame"
            style={
              Object {
                "borderColor": "mediumblue",
              }
            }
          >
            <svg>
              abstract.svg
            </svg>
          </div>
        </div>
        <button
          className="ExhibitionbuttonRight"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "lightskyblue",
              "borderColor": "mediumblue",
              "color": "mediumblue",
            }
          }
        >
          Next &gt;
        </button>
      </div>
      <div
        className="ExtraButtons"
      >
        <div>
          <button
            className="ExtraButton"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "lightskyblue",
                "borderColor": "mediumblue",
                "color": "mediumblue",
              }
            }
          >
             
            <svg>
              pause.svg
            </svg>
             
          </button>
          Rotation
        </div>
        <div>
          <button
            className="ExtraButton"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "lightskyblue",
                "borderColor": "mediumblue",
                "color": "mediumblue",
              }
            }
          >
            <svg>
              unstar.svg
            </svg>
             
          </button>
          Favorite
        </div>
        <div>
          <button
            className="ExtraButton"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "lightskyblue",
                "borderColor": "mediumblue",
                "color": "mediumblue",
              }
            }
          >
            <svg>
              audio.svg
            </svg>
             
          </button>
          Sound
        </div>
      </div>
      <div>
        <audio
          autoPlay={true}
          loop={true}
          muted={false}
          src="Abstract.mp3"
        />
      </div>
    </div>
    <h2
      className="about"
    >
      About the exhibition
    </h2>
    <h4
      className="text"
    >
      Info
    </h4>
    <p
      className="text"
    >
      This exhibition is made to combine different artistic mediums to a large art piece. The exhibition contains eight art pieces with a picture made in SVG, a title, sound and the first line from a poem randomly collected from
       
      <a
        href="https://poetrydb.org/index.html"
        rel="noopener noreferrer"
        target="_blank"
      >
        poetryDB.
      </a>
    </p>
    <h4
      className="text"
    >
      Additions
    </h4>
    <p
      className="text"
    >
      By user request we added the possibility to save a art piece. You can do so by favoriting an image by clicking the star button. Next time you look at the picture the star button is marked and the poem is the same as the saved one. If you unfavorit a new poem wil be added next time.
    </p>
    <p
      className="text"
    >
      Rotation is default 10s and can be changed in settings. The change will be saved until you close your browser. Apply be pressing enter after writing in a number from 1s to how high you want in seconds. You can pause with pause-button and it will automatically pause if you change picture manually.
    </p>
    <p
      className="text"
    >
      In settings you can change color theme and it will stay the same until you refresh the page.
    </p>
    <p
      className="text"
    >
      It is added fitting music to the art. If you want to enjoy the art in silence you can press the sound-button and mute.
    </p>
    <ul>
      <b
        className="text"
      >
        Credits for the music and sound
      </b>
      <li
        className="text"
      >
        Abstract: Piano Horror - Francisco Alvear
         
        <a
          href="https://mixkit.co/free-stock-music/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Link to song
        </a>
      </li>
      <li
        className="text"
      >
        Car: Car sounds from 1:00 - 1:08
         
        <a
          href="https://www.youtube.com/watch?v=hWrwloX6H9U&ab_channel=19Bozzy92"
          rel="noopener noreferrer"
          target="_blank"
        >
          Link to sounds
        </a>
      </li>
      <li
        className="text"
      >
        Desert: Sun and His Daughter - Eugenio Mininni
         
        <a
          href="https://mixkit.co/free-stock-music/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Link to song
        </a>
      </li>
      <li
        className="text"
      >
        Duff: Comical - Ahjay Stelino
         
        <a
          href="https://mixkit.co/free-stock-music/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Link to song
        </a>
      </li>
      <li
        className="text"
      >
        Head: Hip Hop 02 - Lily J
         
        <a
          href="https://mixkit.co/free-stock-music/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Link to song
        </a>
      </li>
      <li
        className="text"
      >
        House: Spirit in the Woods - Alejandro Maga√±a (A. M.)
         
        <a
          href="https://mixkit.co/free-stock-music/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Link to song
        </a>
      </li>
      <li
        className="text"
      >
        ProgPils: Games Worldbeat - Bernardo R.
         
        <a
          href="https://mixkit.co/free-stock-music/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Link to song
        </a>
      </li>
      <li
        className="text"
      >
        TwoHouse: Sugar Baby - Dock Boggs
         
        <a
          href="http://www.openmusicarchive.org/search.php"
          rel="noopener noreferrer"
          target="_blank"
        >
          Link to song
        </a>
      </li>
    </ul>
  </div>
</div>
`;
